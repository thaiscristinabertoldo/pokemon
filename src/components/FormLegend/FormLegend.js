import React, { memo } from 'react'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import {
  FormLegendStyled,
  IconStyled,
  LegendGroupStyled,
  LabelTitleStyled,
  LabelSubTitleStyled,
} from './formlegend.css'

export const FormLegend = memo(({ icon, title, subtitle }) => {
  return (
    <FormLegendStyled>
      {icon && (
        <IconStyled>
          <FontAwesomeIcon icon={icon} size="2x" swapOpacity data-testid="icon" />
        </IconStyled>
      )}

      <LegendGroupStyled>
        {title && <LabelTitleStyled>{title}</LabelTitleStyled>}
        {subtitle && <LabelSubTitleStyled>{subtitle}</LabelSubTitleStyled>}
      </LegendGroupStyled>
    </FormLegendStyled>
  )
})
