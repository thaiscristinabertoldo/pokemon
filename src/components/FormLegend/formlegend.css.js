import styled from 'styled-components'

import { COLORS } from '../../constants'

const FormLegendStyled = styled.div`
  display: flex;
  align-items: center;
  margin: 10px 0;
`

const IconStyled = styled.div`
  margin-right: 10px;
  opacity: 0.4;
`

const LegendGroupStyled = styled.div`
  display: flex;
  flex-direction: column;
`

const LabelTitleStyled = styled.label`
  font-weight: bold;
  color: ${COLORS.mediumGrey};
  font-size: 18px;
`

const LabelSubTitleStyled = styled.label`
  color: ${COLORS.mediumGrey};
  font-size: 12px;
`

export { FormLegendStyled, IconStyled, LegendGroupStyled, LabelTitleStyled, LabelSubTitleStyled }
