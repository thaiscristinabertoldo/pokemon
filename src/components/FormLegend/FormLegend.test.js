import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { faIdBadge } from '@fortawesome/free-solid-svg-icons'

import { FormLegend } from './FormLegend'

afterEach(cleanup)

describe('FormLegend', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<FormLegend />)

    expect(asFragment(<FormLegend />)).toMatchSnapshot()
  })

  it('should render a title', () => {
    const { getByText } = render(<FormLegend title="Text title" />)
    expect(getByText('Text title')).toBeTruthy()
  })

  it('should render a subtitle', () => {
    const { getByText } = render(<FormLegend subtitle="Text subtitle" />)
    expect(getByText('Text subtitle')).toBeTruthy()
  })

  it('should render a icon', () => {
    const { getByTestId } = render(<FormLegend icon={faIdBadge} />)
    expect(getByTestId('icon')).toBeTruthy()
  })
})
