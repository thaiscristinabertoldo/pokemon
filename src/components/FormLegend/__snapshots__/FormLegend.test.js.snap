// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLegend should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm ibOowz"
  >
    <div
      class="sc-AxiKw jWoTjM"
    />
  </div>
</DocumentFragment>
`;
